atproto/identity/handle.go

```diff
@@ -0,0 +1,101 @@
+package identity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+// Does not cross-verify, just does the handle resolution step.
+func ResolveHandleDNS(ctx context.Context, handle syntax.Handle) (syntax.DID, error) {
+	// TODO: timeout
+	// TODO: mechanism to control NDS better; context? separate method?
```

Comment:
[the mikeg DNS package](https://github.com/miekg/dns) is a godsend here, the `net` DNS stuff is very basic

Reply:
I looked at this but didn't want to pull in a dependency in this initial implementation. I did: 

- expose the `net.Resolver`, which can be used to point to a specific DNS server easily, or be swapped out with a not-too-crazy interface implementation wrapping something else
- implemented a basic/partial authoritative lookup with the stdlib net DNS (not as complete as your implementation)

and open to iterating on the API/implementation.